lib/contract_interact/pricer.js

```diff
@@ -0,0 +1,357 @@
+      this.contractAddress,
+      encodedABI,
+      senderAddress,
+      senderPassphrase,
+      { gasPrice: gasPrice, gas: GAS_LIMIT }
+    ).then(function(transactionHash) {
+      return Promise.resolve(responseHelper.successWithData({transactionHash: transactionHash}));
+    });
+  },
+
+
+  /**
+  * Pay
+  *
+  * @return {Promise}
+  *
+  */
+  pay: function (
+    senderAddress,
+    senderPassphrase,
+    beneficiaryAddress,
+    transferAmount,
+    commissionBeneficiaryAddress,
+    commissionAmount,
+    currency,
+    intendedPricePoint,
+    gasPrice) {
+
+    if (gasPrice === undefined || gasPrice === '' || gasPrice == 0 ) {
+      return Promise.reject('GasPrice is mandatory');
+    }
+    if (transferAmount < 0) {
+      return Promise.reject('Transfer amount cannot be negetive');
+    }
+    if (commissionAmount < 0) {
+      return Promise.reject('Commission amount cannot be negetive');
+    }
+    helper.assertAddress(senderAddress);
+    if (transferAmount > 0 || (beneficiaryAddress !== undefined && beneficiaryAddress !== '')) {
+      helper.assertAddress(beneficiaryAddress);
+    }
+    if (commissionAmount > 0 || (commissionBeneficiaryAddress !== undefined && commissionBeneficiaryAddress !== '')) {
+      helper.assertAddress(commissionBeneficiaryAddress);
+    }
+
+    const transactionObject = currContract.methods.pay(
+      beneficiaryAddress,
+      transferAmount,
+      commissionBeneficiaryAddress,
+      commissionAmount,
+      web3RpcProvider.utils.asciiToHex(currency),
+      intendedPricePoint);
+
+    const encodedABI = transactionObject.encodeABI();
+
+    return helper.sendTxAsyncFromAddr(
+      web3RpcProvider,
+      this.contractAddress,
+      encodedABI,
+      senderAddress,
+      senderPassphrase,
+      { gasPrice: gasPrice, gas: GAS_LIMIT }
+    ).then(function(transactionHash) {
+      return Promise.resolve(responseHelper.successWithData({transactionHash: transactionHash}));
+    });
+
+  },
+
+
+  /**
+  * Get current price point and token decimal for the price oracle for the give currency
+  *
+  * @return {Promise}
+  *
+  */
+  getPricePoint: async function (currency) {
+    if (currency === undefined || currency === '') {
+      return Promise.reject('Currency is mandatory');
+    }
+    const transactionObject = currContract.methods.getPricePoint(web3RpcProvider.utils.asciiToHex(currency));
+    const encodedABI = transactionObject.encodeABI();
+    const transactionOutputs = helper.getTransactionOutputs(transactionObject);
+    //const response = await helper.call(web3RpcProvider, this.contractAddress, encodedABI, {}, transactionOutputs);
+    //return Promise.resolve(responseHelper.successWithData({pricePoint: response[0]}));
+    return helper.call(
+      web3RpcProvider,
+      this.contractAddress,
+      encodedABI,
+      {},
+      transactionOutputs)
+      .then(function(response) {
+        return Promise.resolve(responseHelper.successWithData({pricePoint: response[0]}));
```

Comment:
I syntax i have followed in case of multiple returns in method is wrapped promise.

Reply:
Will be done in caching and notification integration